cli/cook/querying.py

```diff
@@ -72,6 +67,30 @@ def satisfy_pred():
     return entities
 
 
+def partition(l, n):
+    """Yield successive n-sized chunks from l"""
+    for i in range(0, len(l), n):
+        yield l[i:i + n]
+
+
+def query_cluster(cluster, uuids, pred, timeout, interval, make_request_fn, entity_type):
+    """Delegates to __query_cluster in batches of at most 100 UUIDs and combines the results"""
+    if len(uuids) == 0:
+        return []
+
+    # Cook will give us back two copies if the user asks for the same UUID twice, e.g.
+    # $ cs show d38ea6bd-8a26-4ddf-8a93-5926fa2991ce d38ea6bd-8a26-4ddf-8a93-5926fa2991ce
+    # Prevent this by calling distinct:
+    uuids = __distinct(uuids)
+
+    entities = []
+    query_batch_size = 100
+    for uuid_batch in partition(uuids, query_batch_size):
```

Comment:
Should this be inside a try-catch block and the exceptions aggregated?

Reply:
`make_data_request` (which is called by `__query_cluster`) takes care of a couple types of exceptions:

```python
def make_data_request(cluster, make_request_fn):
    """
    Makes a request (using make_request_fn), parsing the
    assumed-to-be-JSON response and handling common errors
    """
    try:
        resp = make_request_fn()
        if resp.status_code == 200:
            return resp.json()
        else:
            return []
    except requests.exceptions.ConnectionError as ce:
        logging.exception(ce)
        raise Exception(f'Encountered connection error with {cluster["name"]} ({cluster["url"]}).')
    except requests.exceptions.ReadTimeout as rt:
        logging.exception(rt)
        raise Exception(f'Encountered read timeout with {cluster["name"]} ({cluster["url"]}).')
    except IOError as ioe:
        logging.exception(ioe)
        return []
    except json.decoder.JSONDecodeError as jde:
        logging.exception(jde)
        return []
```

Any others I think we actually want to bail out of the whole operation for.